src/fmu/dataio/_objectdata_provider.py

```diff
@@ -812,13 +780,13 @@ def _derive_from_existing(self) -> None:
         # TODO: Clean up types below.
         self.time0, self.time1 = parse_timedata(self.meta_existing["data"])  # type: ignore
 
-    def _process_content(self) -> tuple[str, dict | None]:
+    def _process_content(self) -> tuple[str | dict, dict | None]:
         """Work with the `content` metadata"""
 
         # content == "unset" is not wanted, but in case metadata has been produced while
         # doing a preprocessing step first, and this step is re-using metadata, the
         # check is not done.
-        if self.dataio._usecontent == "unset" and (
+        if self.dataio._usecontent == "unset" and (  # type: ignore[comparison-overlap]
```

Comment:
Will clean this up in a followup PR, needs a bit more work. I do not want to add it to this PR.